debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/connection/PostgresReplicationConnection.java

```diff
@@ -125,12 +144,29 @@ protected void initPublication() {
                         Long count = rs.getLong(1);
                         if (count == 0L) {
                             LOGGER.info("Creating new publication '{}' for plugin '{}'", publicationName, plugin);
-                            // Publication doesn't exist, create it.
-                            // todo: DBZ-766 - Change this to be restricted based on configured whitelist tables?
-                            // For situations where no publication exists, we likely cannot create it for all tables.
-                            // This is because postgres requires certain super user permissions to use "ALL TABLES".
-                            // We should restrict this to the configured tables here.
-                            stmt.execute(String.format("CREATE PUBLICATION %s FOR ALL TABLES;", publicationName));
+                            switch (publicationAutocreateMode) {
+                                case DISABLED:
+                                    throw new ConnectException("Publication autocreation is disabled, please create one and restart the connector.");
+                                case ALL_TABLES:
+                                    createPublicationStmt = String.format("CREATE PUBLICATION %s FOR ALL TABLES;", publicationName);
+                                    LOGGER.info("Creating Publication with statement '{}'", createPublicationStmt);
+                                    // Publication doesn't exist, create it.
+                                    stmt.execute(createPublicationStmt);
+                                    break;
+                                case FILTERED:
+                                    try {
+                                        Set<TableId> tablesToCapture = determineCapturedTables();
+                                        String tableFilterString = tablesToCapture.stream().map(TableId::toDoubleQuotedString).collect(Collectors.joining(", "));
+                                        createPublicationStmt = String.format("CREATE PUBLICATION %s FOR TABLE %s;", publicationName, tableFilterString);
+                                        LOGGER.info("Creating Publication with statement '{}'", createPublicationStmt);
+                                        // Publication doesn't exist, create it but restrict to the tableFilter.
+                                        stmt.execute(createPublicationStmt);
+                                    }
+                                    catch (Exception e) {
+                                        LOGGER.error("Unable to create publication for ");
```

Comment:
There seems to be missing something after "for". More importantly, though, did you consciously decide to not propagate that exception? I think it'd be better to just let it bubble up (wrapped in a `ConnectException` with some contextual information), thus stopping connector start-up.

Reply:
Actually it looks like I messed something up, I do want to through a connect exception and provide a better error message!